corehq/apps/app_manager/suite_xml.py

```diff
@@ -1696,6 +1697,21 @@ def menus(self):
                 }
                 if self.id_strings.menu_root(module):
                     menu_kwargs['root'] = self.id_strings.menu_root(module)
+
+                if self.app.enable_module_filtering and getattr(module, 'module_filter', None):
+                    PREFIX = './'
+                    USER_REF = 'user'
+                    DATA_REF = 'data'
+
+                    relevant = module.module_filter
+                    if module.module_filter.startswith("{}{}/".format(PREFIX, USER_REF)):
+                        relevant = session_var(relevant[len("{}{}/".format(PREFIX, USER_REF)):],
+                                               subref=USER_REF)
+                    elif module.module_filter.startswith("{}{}/".format(PREFIX, DATA_REF)):
+                        relevant = session_var(relevant[len("{}{}/".format(PREFIX, DATA_REF)):])
```

Comment:
I think we should take a similar approach to form filtering by just expanding the 'dot' into a session reference. That will allow more complex expressions e.g.
```
format-date(date(./user/data/dob), "%n%d") = format-date(now(), '%n%d')
-->
format-date(date(instance('commcaresession')/session/user/data/dob), "%n%d") = format-date(now(), '%n%d')
```